### ISSUE TITLE
prefer-arrow-callback should not be triggered when a function uses the arguments object.

### ISSUE THREAD

**Issue by qubyte** [eslint/eslint#4095]
Arrow functions [do not have their own lexical arguments object](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions#Lexical_arguments), instead inheriting it from the enclosing scope.

A simple example:

```javascript
function getSomeData(callback) {
  callback(1, 2, 3);
}

getSomeData(function() {
  console.log(arguments);
});
```

This triggers the prefer-arrow-callback rule. If the switch to the arrow function is made:

```javascript
getSomeData(() => {
  console.log(arguments); // Arrow functions have no arguments object.
});
```

the value of `arguments` is not that which was intended, and will even throw an error in the global scope.

tl;dr The prefer-arrow-callback rule should also check for the use of `arguments`.

This behaviour seen in v1.6.0.

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by michaelficarra** [comment]
`new.target` as well, I guess.

**Comment by nzakas** [comment]
Probably the same for all of these:

* `arguments`
* `super`
* `new.target`

**Comment by mysticatea** [comment]
I will try to fix this.
